meson.build

```diff
@@ -29,29 +33,43 @@ else
     endif
 endif
 
-curl_dep = dependency('libcurl')
-
-if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
-    argplib = cc.find_library('argp')
+if build_machine.system() == 'windows'
+    curl_dep = dependency('curl', modules: 'CURL::libcurl')
 else
-    argplib = dependency('', required : false)
+    curl_dep = dependency('libcurl')
+endif
+
+if build_machine.system() != 'windows'
+    if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
+       argplib = cc.find_library('argp')
+    else
+       argplib = dependency('', required : false)
+    endif
 endif
 
 if get_option('with-openssl') == 'disabled'
     openssl_dep = dependency('', required : false)
 else
-    openssl_dep = dependency('openssl', required : get_option('with-openssl') == 'enabled')
+    if build_machine.system() == 'windows'
+        openssl_dep = dependency('openssl', modules : ['OpenSSL::SSL', 'OpenSSL::Crypto'], required : get_option('with-openssl') == 'enabled')
```

Comment:
Can you please verify that the `modules` section is required to build in Windows? If not, can we please remove it?

Reply:
There will be a warning or info to tell you about the different modules available IIRC.
CMake exposes multiple "targets" that one can "link" against (that includes libraries & headers). These modules are a way to select the ones we need. 

PS for Linux / macOS (pkg-config) dependency finding nothing has changed).